src/connection.js

```diff
@@ -567,7 +570,8 @@ class Connection extends EventEmitter {
 
   clearRequestTimer() {
     if (this.requestTimer) {
-      return clearTimeout(this.requestTimer);
+      clearTimeout(this.requestTimer);
+      this.requestTimer = undefined;
```

Comment:
> Remove the call to clearTimeout under SENT_CLIENT_REQUEST:message. We don't need that now.

I would like to keep the call to clearRequestTimer() there, to guarantee that the timer is also stopped when a 'message' event is received without a 'data' event. It doesn't hurt to have a redundant call to clearRequestTimer(). Another solution would be to place it in the exit event of the SENT_CLIENT_REQUEST state.

> This thread seems to suggest 'void 0' is safer

This StackOverflow entry is old. Nobody uses "void 0" any more with JavaScript ES6. The initial value of this.requestTimer is `undefined`, so it's appropriate to reset it to `undefined`.

> If we like 'undefined' I'll send a PR to change all 'void 0' s to undefined for consistency. Otherwise it'll keep bugging me :-)

It would be good to clean up the source code, replace `void 0`, remove unnecessary return statements which probably also stem from the CoffeScript conversion, replace `inherit` calls with the new class syntax, convert the remaining CoffeScript files, etc. etc. But keep in mind that the Tedious package consists of 15,300 lines of source code. Until after a complete overhaul, we will have a mix of old and new language constructs. To really clean up the code base, we probably need a sponsor...